### PR TITLE
docs: add Swell to the integration list and rewrite the `Extending GraphQL queries` doc

### PR DESCRIPTION
**Pull request by filipsobol**

### CODE REVIEW

File: packages/core/docs/README.md [hunk 1]
```diff
@@ -27,12 +27,6 @@ _Vue Storefront_ is a ___platform-agnostic e-commerce PWA frontend framework___
   <figcaption style="font-size: 0.9rem">(Click to zoom)</figcaption>
 </figure>
 
-That's a mouthful, so let's break it down:
- - __platform-agnostic__ - we made it possible to work with any platform and service you already use, as long as it has an API like REST or GraphQL.
- - __e-commerce__ - today's shops are much more than just products and carts. That's why we made it easy to integrate other types of services, such as helper service for ERP, versatile search features for PIM, portable checkout for 3rd party payment kiosk, and more.
- - __PWA__ - it's the technology of the future, designed to give the best performance on any device, with native-like features to satisfy your customer's needs.
- - __frontend framework__ - _Vue Storefront_ is a set of modular features glued together using _interfaces_ and _factories_ and powered by [Nuxt.js](https://nuxtjs.org/).
-
```
Comment (VitorLuizC) [inline packages/core/docs/README.md:35]: (Out of the scope for this PR)
Maybe we can have a glossary page with those terms and their meanings.

Comment (filipsobol) [reply packages/core/docs/README.md:35]: It's already on our roadmap 😊

File: packages/core/docs/composition/extending-graphql-queries.md [hunk 2]
```diff
@@ -4,90 +4,99 @@
 The application does not reload automatically after saving the changes in Server Middleware. Due to this, you have to restart the application manually. We are working on enabling Hot Reloading in future updates.
 :::
 
-If your integration uses GraphQL API, you may want to change the default query or mutation sent. That's quite a common case for fetching additional or custom fields. Vue Storefront provides the mechanism for this called "custom queries".
+If given e-commerce integration uses GraphQL API, you may want to change the default queries or mutations sent to it from the Vue Storefront application. It's possible for selected requests using "custom queries".
 
-Since the communication with the API goes through our middleware, all queries also are defined there. To customize or even entirely override the original (default) queries, you need to follow two steps.
+Custom queries allow you to modify or even entirely replace the default GraphQL queries and mutations that the integration uses out of the box. To do this, follow the steps above.
```
Comment (dawid-ziobro) [inline packages/core/docs/composition/extending-graphql-queries.md:9]:
```suggestion
Custom queries allows you to modify or even entirely replace the default GraphQL queries and mutations that the integration uses out of the box. To do this, follow the steps below.
```

File: packages/core/docs/composition/extending-graphql-queries.md [hunk 3]
```diff
@@ -4,90 +4,99 @@
...
+If the method doesn't support extending, you can instead replace the whole API endpoint called by it using the `extendApiMethod` described on the [Extending integrations](/integrate/extending-integrations.html) page.
 
-In the example above, we change the `products` query with our custom query named `my-products-query`. Additionally, the `metadata` field holds additional parameters about the product we seek for. As a second step, we need to define `my-products-query` query.
+### Step 2: Register custom query
 
-Each custom query lives in the `middleware.config.js`, so it's the place where we should define `my-products-query`.
+Custom queries are registered individually for every integration in the `middleware.config.js` file. They are function represented by the [CustomQueryFn](/reference/api/core.customqueryfn.html) type.
 
-Custom query functions have the arguments:
+These functions accept one object parameter with the following properties:
```
Comment (dawid-ziobro) [inline packages/core/docs/composition/extending-graphql-queries.md:25]:
```suggestion
These functions accepts one object parameter with the following properties:
```

File: packages/core/docs/composition/extending-graphql-queries.md [hunk 4]
```diff
@@ -4,90 +4,99 @@
...
-          query: `
-            query products($where: String) {
-              products(where: $where) { /* ... */ }
-            }
-          `,
-          variables,
-        })
-      }
-    }
+Object keys are the names of the default queries. To get them, go to the documentation of the [e-commerce integration](/integrations) of your choice and look for the page describing the composable, then the section for the method you want to override and find the `customQuery` key.
```
Comment (dawid-ziobro) [inline packages/core/docs/composition/extending-graphql-queries.md:67]: That sentence is so long and hard to understand.

File: packages/core/docs/composition/extending-graphql-queries.md [hunk 5]
```diff
@@ -4,90 +4,99 @@
...
+
+Object values are the names of the custom queries you defined in the `middleware.config.js` file.
```
Comment (dawid-ziobro) [inline packages/core/docs/composition/extending-graphql-queries.md:69]:
```suggestion
Object values are the names of the custom queries that you defined in the `middleware.config.js` file.
```

**Review by dawid-ziobro — APPROVED** [review]

**Review by dawid-ziobro — APPROVED** [review]

**Comment by nbpalomino** [comment]
Is there a implementation for Swell???

**Pull request merged**